lib/py-ios-device-client.js

```diff
@@ -0,0 +1,80 @@
+import { exec } from 'teen_process';
+import { fs, util, tempDir } from 'appium-support';
+import log from './logger';
+import path from 'path';
+
+// https://github.com/YueChen-C/py-ios-device
+
+const BINARY_NAME = 'pyidevice';
+
+class Pyidevice {
+  constructor (udid) {
+    this.udid = udid;
+    this.binaryPath = null;
+  }
+
+  async assertExists (isStrict = true) {
+    if (this.binaryPath) {
+      return true;
+    }
+
+    try {
+      this.binaryPath = await fs.which(BINARY_NAME);
+      return true;
+    } catch (e) {
+      if (isStrict) {
+        throw new Error(`${BINARY_NAME} binary cannot be found in PATH. ` +
+          `Please make sure it is installed. Visit https://github.com/YueChen-C/py-ios-device for ` +
+          `more details.`);
+      }
+      return false;
+    }
+  }
+
+  async execute (args, ensureFormat = true) {
+    await this.assertExists();
+
+    const finalArgs = ['--udid', this.udid, ...args];
+    if (ensureFormat) {
+      finalArgs.push('--format', 'json');
+    }
+    log.debug(`Executing ${util.quote(this.binaryPath, finalArgs)}`);
```

Comment:
mm, this quote function seems raising `TypeError: xs.map is not a function`

Reply:
fixed. The argument must be an array there.

Reply:
```
[debug] [XCUITest] Executing command 'execute'
[debug] [W3C (d7d694d2)] Encountered internal error running command: TypeError: Cannot read properties of undefined (reading 'replace')
[debug] [W3C (d7d694d2)]     at /Users/kazuaki/GitHub/appium-xcuitest-driver/node_modules/shell-quote/index.js:4:25
[debug] [W3C (d7d694d2)]     at Array.map (<anonymous>)
[debug] [W3C (d7d694d2)]     at exports.quote (/Users/kazuaki/GitHub/appium-xcuitest-driver/node_modules/shell-quote/index.js:2:15)
[debug] [W3C (d7d694d2)]     at Object.quote (/Users/kazuaki/GitHub/appium-xcuitest-driver/node_modules/appium-support/lib/util.js:340:10)
[debug] [W3C (d7d694d2)]     at Pyidevice.execute (/Users/kazuaki/GitHub/appium-xcuitest-driver/lib/py-ios-device-client.js:41:25)
[debug] [W3C (d7d694d2)]     at Pyidevice.installProfile (/Users/kazuaki/GitHub/appium-xcuitest-driver/lib/py-ios-device-client.js:72:9)
[debug] [W3C (d7d694d2)]     at XCUITestDriver.mobileInstallCertificate (/Users/kazuaki/GitHub/appium-xcuitest-driver/lib/commands/certificate.js:316:7)
[debug] [W3C (d7d694d2)]     at XCUITestDriver.executeMobile (/Users/kazuaki/GitHub/appium-xcuitest-driver/lib/commands/execute.js:158:10)
[debug] [W3C (d7d694d2)]     at XCUITestDriver.execute (/Users/kazuaki/GitHub/appium-xcuitest-driver/lib/commands/execute.js:32:12)
```

Should be `util.quote([this.binaryPath, ...finalArgs]);`?

Reply:
thanks, fixed